pkg/iniparser/parser.go

```
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+var (
+	ErrSectionNotExists = errors.New("Section doesn't exist")
+	ErrKeyNotExists     = errors.New("cKey doesn't exist")
+)
+
+type Iniparser struct {
+	sections map[string]map[string]string
+}
+
+// a method to load the data from the a string that holds the ini structure into the iniparser
+func (i *Iniparser) LoadFromString(config string) {
+	i.sections = map[string]map[string]string{}
+	lines := strings.Split(config, "\n")
+	section := ""
+	m := map[string]string{}
+
+	for _, line := range lines {
+		line := strings.TrimSpace(line)
+		if len(line) != 0 {
+			if line[0] == '[' {
```

this will parse lines like `[abc` as sections which is incorrect and it should return an error in this case